src/awsHttpHandler/httpHandler.ts

```diff
@@ -0,0 +1,116 @@
+import { hckFetch } from '../index';
+import { requestTimeout } from './requestTimeout';
+import { isEmpty } from './isEmpty';
+
+type LogHandler = (...content: unknown[]) => string;
+
+export type Logger = {
+  log?: LogHandler;
+  info?: LogHandler;
+};
```

Comment:
I think that you should not have `.log()` and rather define one method per level. Consumers that have just a `.log()` method can implement all levels by calling that same function. 

```suggestion
export type Logger = {
  debug?: LogFn;
  info?: LogFn;
  warn?: LogFn;
};
```
Note that a library is supposed to throw errors, not to log them: that's why there is no `error` method.